x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_import_export/trial_license_complete_tier/import_connectors.ts

```diff
@@ -0,0 +1,516 @@
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  describe('@ess @brokenInServerless @skipInQA import action connectors', () => {
+    const CONNECTOR_ID = '1be16246-642a-4ed8-bfd3-b47f8c7d7055';
+    const ANOTHER_CONNECTOR_ID = 'abc16246-642a-4ed8-bfd3-b47f8c7d7055';
+    const CUSTOM_ACTION_CONNECTOR = {
+      id: CONNECTOR_ID,
+      type: 'action',
+      updated_at: '2024-02-05T11:52:10.692Z',
+      created_at: '2024-02-05T11:52:10.692Z',
+      version: 'WzYsMV0=',
+      attributes: {
+        actionTypeId: '.email',
+        name: 'test-connector',
+        isMissingSecrets: false,
+        config: {
+          from: 'a@test.com',
+          service: 'other',
+          host: 'example.com',
+          port: 123,
+          secure: false,
+          hasAuth: false,
+          tenantId: null,
+          clientId: null,
+          oauthTokenUrl: null,
+        },
+        secrets: {},
+      },
+      references: [],
+      managed: false,
+      coreMigrationVersion: '8.8.0',
+      typeMigrationVersion: '8.3.0',
+    };
+
+    beforeEach(async () => {
+      await deleteAllRules(supertest, log);
+      await deleteConnector(supertest, CONNECTOR_ID);
+      await deleteConnector(supertest, ANOTHER_CONNECTOR_ID);
+    });
+
+    describe('overwrite connectors is set to "false"', () => {
+      it('imports a rule with an action connector', async () => {
+        const ndjson = combineToNdJson(
+          getCustomQueryRuleParams({
+            rule_id: 'rule-1',
+            name: 'Rule 1',
+            actions: [
+              {
+                group: 'default',
+                id: CONNECTOR_ID,
+                params: {
+                  message: 'Some message',
+                  to: ['test@test.com'],
+                  subject: 'Test',
+                },
+                action_type_id: '.email',
+                uuid: 'fda6721b-d3a4-4d2c-ad0c-18893759e096',
+                frequency: { summary: true, notifyWhen: 'onActiveAlert', throttle: null },
+              },
+            ],
+          }),
+          CUSTOM_ACTION_CONNECTOR
+        );
+
+        const { body } = await supertest
+          .post(`${DETECTION_ENGINE_RULES_URL}/_import`)
+          .set('kbn-xsrf', 'true')
+          .set('elastic-api-version', '2023-10-31')
+          .attach('file', Buffer.from(ndjson), 'rules.ndjson')
+          .expect(200);
+
+        expect(body).toMatchObject({
+          errors: [],
+          success: true,
+          success_count: 1,
+          rules_count: 1,
+          action_connectors_success: true,
+          action_connectors_success_count: 1,
+          action_connectors_errors: [],
+          action_connectors_warnings: [],
+        });
+
+        expect(await getConnector(supertest, CONNECTOR_ID)).toMatchObject({
+          id: CONNECTOR_ID,
+          name: 'test-connector',
+        });
+      });
+
+      it('DOES NOT import an action connector without rules', async () => {
```

Comment:
So the current behaviour is that the import endpoint return 200, even with `action_connectors_success: true`, but the connector is actually not created?

Sounds like bad UX. Should create a ticket to change this into throwing an error? Or is this expected/documented behaviour?

Reply:
Good point. The thing is the rule import is rule centric. Rule is the main concern. If there are no rules to import it returns success status. If there are extra action connectors (not all action connector are referenced by rule actions) only used by rule action connectors are imported.

Changing the endpoint's behavior may be problematic as it's a public one. But it's worth to reconsider how rule import works.

@banderror do you have any thoughts here?

Reply:
@maximpn My expectation - as a user who doesn't know much about the rule import feature - would be: anything I pass in a `.ndjson` file should be created or updated, if it's valid. I don't get why connectors that are not referenced from rules, should NOT be imported. It's a "clever" sort of logic that I never like in software.

I'd suggest to create a discussion ticket for that.

Reply:
@banderror The discussion ticket is here https://github.com/elastic/security-team/issues/8644